setup.cfg

```diff
@@ -41,6 +41,7 @@ python_requires = >=3.8
 install_requires =
   python-dateutil
   lark
+  numpy
```

Comment:
If we merge #88 first, we'll have to move this to the `pyproject.toml` dependencies.

Reply:
Do we want to limit this to NumPy v2, which was released in May? That was the first major release since 2006. If we do limit it to v2, we should also drop support for Python < 3.10. It looks like NumPy v2 requires Python >= 3.10.

I'm pretty sure that there are no direct Python dependencies, just some optional dependencies for speeding up operations. `pip` should provide pre-built wheels.

NumPy looks like it will add 30-80MB to the payload. For most applications that's not an issue. But maybe we should benchmark the current version of `undate` and the size with `numpy`? The environments that I'm typically space-constrained in are Lambdas. I think there is still a 50MB (zipped) and 250MB (unzipped) limit for Lambda functions, so this could take up a significant chunk of that.

Reply:
I think I'm still leaning towards using `NumPy` given the need to support dates outside of 9999. The other way that we could go is with `struct_time`, which is what `python-edtf` uses. But it's a little clunky and you'd still have to have a shim class (like you add here) or a conversion function.

Reply:
Trying to make NumPy optional seems like a nightmare

Reply:
I was thinking it would be nice to move to Python 3.10+ anyway, it gets us improved python type hinting and 3.9 is getting old enough at this point that it seems reasonable to drop.